src/Dfc.CourseDirectory.WebV2/Features/Providers/ProviderDetails.cs

```
@@ -22,31 +24,41 @@ public class ViewModel
         public string CourseDirectoryStatus { get; set; }
         public int Ukprn { get; set; }
         public string TradingName { get; set; }
+        public bool ShowDisplayName { get; set; }
+        public string DisplayName { get; set; }
         public ProviderType ProviderType { get; set; }
         public bool CanChangeProviderType { get; set; }
     }
 
     public class Handler : IRequestHandler<Query, ViewModel>
     {
         private readonly ICosmosDbQueryDispatcher _cosmosDbQueryDispatcher;
+        private readonly ISqlQueryDispatcher _sqlQueryDispatcher;
         private readonly ICurrentUserProvider _currentUserProvider;
 
         public Handler(
             ICosmosDbQueryDispatcher cosmosDbQueryDispatcher,
+            ISqlQueryDispatcher sqlQueryDispatcher,
             ICurrentUserProvider currentUserProvider)
         {
             _cosmosDbQueryDispatcher = cosmosDbQueryDispatcher;
+            _sqlQueryDispatcher = sqlQueryDispatcher;
             _currentUserProvider = currentUserProvider;
         }
 
         public async Task<ViewModel> Handle(Query request, CancellationToken cancellationToken)
         {
-            var provider = await _cosmosDbQueryDispatcher.ExecuteQuery(new GetProviderById()
+            var cosmosProvider = await _cosmosDbQueryDispatcher.ExecuteQuery(new CosmosQueries.GetProviderById()
             {
                 ProviderId = request.ProviderId
             });
 
-            if (provider == null)
+            var sqlProvider = await _sqlQueryDispatcher.ExecuteQuery(new SqlQueries.GetProviderById()
+            {
+                ProviderId = request.ProviderId
+            });
+
+            if (cosmosProvider == null)
```

Could fail faster? (Before invoking the SqlQuery)